src/__test__/backend/mcode/services/CLQOutcomesService.test.js

```diff
@@ -0,0 +1,105 @@
+
+    let clqService = new CLQOutcomesService({
+        serviceUrl: "http://localhost/outcomes"
+    });
+
+    let patient = new PatientRecord(BreastMainTreatmentDiabetesHypertensionJaneV05);
+   
+    
+    let similarPatientProps = getProps(patient, patient.getActiveConditions()[0])
+    // select everything in the options 
+    _.forIn(similarPatientProps.demographic.options, function (value, key) {
+        value.selected = true
+    });
+    _.forIn(similarPatientProps.pathology.options, function (value, key) {
+        value.selected = true
+    });
+
+    beforeEach(() => {
+        nock.disableNetConnect();
+    })
+
+    afterEach(() => {
+        nock.cleanAll();
+        nock.enableNetConnect(); // just to be sure this doesn't break any other tests anywhere else
+    });
+
+    it("Should be able to create demographics filter", () => {
+        let clqFilter = clqService.buildDemographicsFilter(similarPatientProps);
+        expect(_.isEqual(clqFilter, expectedFilter.demographics)).to.be.true
+    });
+
+    it("Should be able to create tumormarker filter filter", () => {
+        let clqFilter = clqService.buildTumorMakersFilter(similarPatientProps);
+        expect(_.isEqual(clqFilter, expectedFilter.tumorMarkers)).to.be.true
+    });
+
+    it("Should be able to create diagnosis filter", () => {
+        let clqFilter = clqService.buildDiagnosisFilter(similarPatientProps);
+        expect(_.isEqual(clqFilter, expectedFilter.diagnosis)).to.be.true
+
+    });
+
+    it("Should be able to translate results to rows", () => {
+        let clqResults = clqService.generateOutcomeData(response.outcomes.survival.data)
+        expect(_.isEqual(clqResults, rows)).to.be.true
+    });
+
+    it("Should be able to process results", () => {
+        let clqResults = clqService.processResults(response, [{
+            "code": "A",
+            "displayName": "A",
+            "codeSystem": "2.16.840.1.113883.6.88",
+            "codeSystemName": "RXNORM"
+        }], [{
+                "code": "A",
+                "displayName": "A",
+                "codeSystem": "2.16.840.1.113883.6.88",
+                "codeSystemName": "RXNORM"
+            },
+            {
+                "code": "B",
+                "displayName": "B",
+                "codeSystem": "2.16.840.1.113883.6.88",
+                "codeSystemName": "RXNORM"
+            }
+        ])
+        expect(_.isEqual(clqResults, processed)).to.be.true
+    });
+
+    it("Should be able to take a request and return results", (done) => {
+
+        const scope = nock('http://localhost/')
+            .post('/outcomes')
+            .reply(200, response)
+
+        clqService.processSimilarPatientOutcomes(similarPatientProps).then((clqResults) => {
+            expect(_.isEqual(clqResults, processed)).to.be.true
+            scope.done();
+            done();
+        }).catch((err) => {
+            console.log(err);
+            scope.done();
+            done();
```

Comment:
If an exception gets thrown somewhere and caught here, does the test fail? `scope.done()` will fail the test if the URL defined above wasn't hit, but other than that this seems like it will pass even if there was an error.

Reply:
You are correct and upon looking into it there is a bit of an issue with testing.   Ids for the results are generated with _.uniqueId('row_').  This seems to use a global behind the scenes which causes there to be issues with multiple tests running and changing the expected id's of the rows.   Trying to track down a way to deal with it and will post an update.

Reply:
Modification pushed to deal with testing unique identifiers and incorporate failing this particular test when there is an error